internal/client/client.go

```diff
@@ -92,6 +92,13 @@ func WrapDebug(c *http.Client, debug bool) *http.Client {
 	return c
 }
 
+func WrapHTTPClientWithOauth2(ctx context.Context, httpClient *http.Client, clientCredentials *clientcredentials.Config) *http.Client {
+	ctx = context.WithValue(oauth2.NoContext, oauth2.HTTPClient, httpClient)
+	client := oauth2.NewClient(ctx, clientCredentials.TokenSource(ctx))
+	httpClient.Transport = client.Transport
```

Comment:
I am curious why we would need to use the httpClient after we have instructed `oauth2.NewClient` to use it as its base.. Isn't it enough to just return `oauth2.NewClient(...)`?

Reply:
I've pushed a minor update so that it uses [clientCredentials.Client()](https://github.com/golang/oauth2/blob/master/clientcredentials/clientcredentials.go#L66-L68) (which will call `oauth2.NewClient()`) but the question is still valid.

From what I understand from reading [oauth2.NewClient()](https://github.com/golang/oauth2/blob/master/oauth2.go#L342-L352) and investigating the example provided in the previous [comment](https://github.com/go-auth0/auth0/pull/112#issuecomment-628082781), it will overwrite the other users supplied `http.Client` fields like `http.Client.Timeout`. I was unclear if this is common practice so I opted for preserving these supplied fields just in case. To help illustrate further:
```go
client := http.Client{Timeout: Timeout: 2 * time.Second}
newClient := client.WrapHTTPClientWithOauth2(ctx, client, clientCredentials)
```
`newClient.Timeout` will be `nil` but the user had it set.

If I've misunderstood this, let me know. Also let me know if you feel that  `WrapHTTPClientWithOauth2()` should overwrite a user supplied field like `http.Client.Timeout` or preserve it. This is also relevant for the other Wrap functions, which attempt to preserve these fields too.